src/librustdoc/formats/cache.rs

```diff
@@ -215,7 +215,7 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
         // Propagate a trait method's documentation to all implementors of the
         // trait.
         if let clean::TraitItem(ref t) = *item.kind {
-            self.cache.traits.entry(item.def_id.expect_real()).or_insert_with(|| {
+            self.cache.traits.entry(item.def_id.expect_def_id()).or_insert_with(|| {
```

Comment:
As a follow-up PR, I think it might be possible to change `traits` to store `DefId`s as the key directly.

Reply:
Do you mean `traits` to store `ItemId`s?

Reply:
Err I think I was confused. The code is fine as-is.